Firestore/core/test/unit/credentials/firebase_auth_credentials_provider_test.mm

```diff
@@ -153,6 +153,29 @@ - (void)getTokenForcingRefresh:(BOOL)forceRefresh
   });
 }
 
+// Catch race conditions in FirebaseAuthCredentialsProvider::GetToken() when
+// run under thread sanitizer.
+// See https://github.com/firebase/firebase-ios-sdk/issues/10393
+TEST(FirebaseAuthCredentialsProviderTest, GetTokenCalledByAnotherThread) {
+  FIRApp* app = testutil::AppForUnitTesting();
+  FSTAuthFake* auth = [[FSTAuthFake alloc] initWithToken:@"token for fake uid"
+                                                     uid:@"fake uid"];
+  FirebaseAuthCredentialsProvider credentials_provider(app, auth);
+  credentials_provider.InvalidateToken();
+
+  std::thread thread1([&credentials_provider] {
+    credentials_provider.GetToken(
+        [](util::StatusOr<AuthToken> result) { EXPECT_TRUE(result.ok()); });
+  });
+  std::thread thread2([&credentials_provider] {
+    credentials_provider.GetToken(
+        [](util::StatusOr<AuthToken> result) { EXPECT_TRUE(result.ok()); });
+  });
+
+  thread1.join();
+  thread2.join();
```

Comment:
What would happen to this test if we used previous nonatomic implementation?

Reply:
Changing `atomic` back to `nonatomic` causes the Thread Sanitizer tests to fail with a race condition. I don't recall exactly what the race condition was though. So, I've pushed up a commit to this PR, https://github.com/firebase/firebase-ios-sdk/pull/10403/commits/e9cc0f9a6d7c32ab14ce6a9927b1a3e06ca4cc18, that changes `atomic` back to `nonatomic`. We'll see how/if this test fails then I'll reply back to this comment.